### ISSUE TITLE
Percent symbol is not escaped in TickLabel

### ISSUE THREAD

**Issue by danjjl** [matlab2tikz/matlab2tikz#710]
# Minimal MATLAB working example:
```matlab
plot(1:10)
set(gca,'XTick',5)
set(gca,'XTickLabel','50%')
matlab2tikz('bug.tex')
```

# matlab2tikz .tex output
```tex
...
xtick={5},
xticklabels={{50%}},
...
```

# Expected .tex output
```tex
...
xtick={5},
xticklabels={{50\%}},
...
```

The percent symbol should be escaped.

*(Using commit 33667a0829a3223c8944cc502505b78e3496c8d7)*

**Comment by egeerardyn** [comment]
Thank you for reporting this bug and having a nice minimal working example.
I don't think it should be too hard to fix.

In the meanwhile, you can work around the issue by doing `set(gca,'XTickLabel','50\%')`, but I suspect that by the next release this bug should be gone.

**Comment by PeterPablo** [comment]
Nice catch @danjjl! Tick labels were not escaped. Here is the diff, if you want to fix your version right away:
```diff
diff --git a/src/matlab2tikz.m b/src/matlab2tikz.m
index 4a1fda4..9e4b696 100644
--- a/src/matlab2tikz.m
+++ b/src/matlab2tikz.m
@@ -1278,11 +1278,14 @@ function [options] = getAxisTicks(m2t, handle, axis, options)
 
     keywordTickLabelMode = [upper(axis), 'TickLabelMode'];
     tickLabelMode = get(handle, keywordTickLabelMode);
-    keywordTickLabel = [upper(axis), 'TickLabel'];
-    tickLabels = cellstr(get(handle, keywordTickLabel));
     if strcmpi(tickLabelMode, 'auto') && ~m2t.cmdOpts.Results.strict
         pgfTickLabels = [];
     else % strcmpi(tickLabelMode,'manual') || m2t.cmdOpts.Results.strict
+        keywordTickLabel = [upper(axis), 'TickLabel'];
+        interpreter = 'none'; % MATLAB only supports 'none'
+        tickLabels = cellstr(get(handle, keywordTickLabel))
+        tickLabels = prettyPrint(m2t, tickLabels, interpreter)
+
         keywordScale = [upper(axis), 'Scale'];
         isAxisLog = strcmpi(getOrDefault(handle,keywordScale, 'lin'), 'log');
         [pgfTicks, pgfTickLabels] = ...
```

**Comment by egeerardyn** [comment]
@PeterPablo Thank you for hunting this down. Could you file a PR for this as well?

**Comment by PeterPablo** [comment]
I was being lazy... PR is at #711. Let's see whether you ask me for an adapted ACID test ;-)

**Comment by egeerardyn** [comment]
I think with #703 we should have a test case that covers this (along with a lot of other cases), but for now I'm happy with just a PR to fix this :+1:

**Comment by PeterPablo** [comment]
Great and agreed. The PR is ready for review. Should be rather easy, as all current tests pass :-)

**Comment by egeerardyn** [comment]
This is fixed by #711. So if you use the current version in `develop`, it will work as expected. Otherwise, you will have to wait for the 1.1 release.

**Comment by danjjl** [comment]
Thank you.
You guys were efficient!
